trackintel/io/postgis.py

```diff
@@ -22,348 +56,259 @@ def read_positionfixes_postgis(conn_string, table_name, geom_col="geom", *args,
         Further arguments as available in GeoPanda's GeoDataFrame.from_postgis().
 
     **kwargs
-        Further arguments as available in GeoPanda's GeoDataFrame.from_postgis().
+        Further keyword arguments as available in GeoPanda's GeoDataFrame.from_postgis().
 
     Returns
     -------
     GeoDataFrame
         A GeoDataFrame containing the positionfixes.
-    """
-    engine = create_engine(conn_string)
-    conn = engine.connect()
-    try:
-        pfs = gpd.GeoDataFrame.from_postgis("SELECT * FROM %s" % table_name, conn, geom_col=geom_col, *args, **kwargs)
-    finally:
-        conn.close()
 
-    # assert validity of positionfixes
-    pfs.as_positionfixes
-    return pfs
+    Examples
+    --------
+    >>> pfs = ti.io.read_postifionfixes_postgis("SELECT * FROM postionfixes", con, geom_col="geom")
+    """
+    pfs = gpd.GeoDataFrame.from_postgis(sql, con, geom_col, *args, **kwargs)
+    return ti.io.read_positionfixes_gpd(pfs, geom_col=geom_col)
 
 
+@_handle_con_string
 def write_positionfixes_postgis(
-    positionfixes, conn_string, table_name, schema=None, sql_chunksize=None, if_exists="fail"
+    positionfixes, name, con, schema=None, if_exists="fail", index=True, index_label=None, chunksize=None, dtype=None
 ):
-    """Stores positionfixes to PostGIS. Usually, this is directly called on a positionfixes
-    DataFrame (see example below).
+    positionfixes.to_postgis(name, con, schema, if_exists, index, index_label, chunksize, dtype)
```

Comment:
Here we should use keyword arguments to tunnel these many arguments to prevent future errors.